File: LeeLibrary/scroll/src/main/java/jv/lee/scroll/adapter/SimpleItem.java
package jv.lee.scroll.adapter;

import android.widget.TextView;

import com.lee.library.adapter.LeeViewHolder;
import com.lee.library.adapter.listener.LeeViewItem;

import jv.lee.scroll.R;

/**
 * @author jv.lee
 * @date 2019-09-24
 * @description
 */
public class SimpleItem implements LeeViewItem<String> {
    @Override
    public int getItemLayout() {
        return R.layout.item_text;
    }

    @Override
    public boolean openClick() {
        return true;
    }

    @Override
    public boolean openShake() {
        return true;
    }

    @Override
    public boolean isItemView(String entity, int position) {
        return position != 10;
    }

    @Override
    public void convert(LeeViewHolder holder, String entity, int position) {
        ((TextView)holder.getView(R.id.tv_text)).setText(entity);
    }
}

File: KotlinLibrary/app/src/main/java/com/lee/app/view/adapter/SimpleItem.kt
package com.lee.app.view.adapter

import android.support.v4.content.ContextCompat
import android.view.View
import android.widget.TextView
import com.lee.app.R
import com.lee.library.adapter.LeeViewHolder
import com.lee.library.adapter.listener.LeeViewItem

/**
 * @author jv.lee
 * @date 2019/9/17.
 * @description
 */
class SimpleItem : LeeViewItem<Int> {


    override fun getItemLayout(): Int {
        return R.layout.item_text
    }

    override fun openClick(): Boolean {
        return true
    }

    override fun openShake(): Boolean {
        return true
    }

    override fun isItemView(entity: Int?, position: Int): Boolean {
        return entity != null
    }

    override fun convert(holder: LeeViewHolder?, entity: Int?, position: Int) {
        (holder?.getView<View>(R.id.tv_text) as TextView).text =
            "this is item text position $entity"
        if (position % 2 == 0) {
            holder.convertView.setBackgroundColor(
                ContextCompat.getColor(
                    holder.convertView.context,
                    android.R.color.holo_blue_bright
                )
            )
        } else {
            holder.convertView.setBackgroundColor(
                ContextCompat.getColor(
                    holder.convertView.context,
                    android.R.color.holo_orange_light
                )
            )
        }
    }

    override fun openRecycler(): Boolean {
        return false
    }

    override fun viewRecycled(holder: LeeViewHolder?, entity: Int?, position: Int) {
    }


}